### PR TITLE
MES-4704-fault count now supports sub category Cs

### PR DESCRIPTION
**Pull request by amobarak**
## Description

Fault count implemented for sub category Cs, refactored slightly.

## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [ ] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

**Review by TomBillingtonUK — COMMENTED** [review]
You have only unit tested the getDrivingFaultSumCountCat for the subcategories, there's a lot of other functions added which need unit testing and should be written

File: src/shared/helpers/faults.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { pickBy, endsWith, sumBy } from 'lodash';
+import { CompetencyOutcome } from '../models/competency-outcome';
+
```
Comment (TomBillingtonUK) [inline src/shared/helpers/faults.ts:3]: this file needs unit tests

**Comment by amobarak** [comment]
> You have only unit tested the getDrivingFaultSumCountCat for the subcategories, there's a lot of other functions added which need unit testing and should be written

No, just no

**Comment by amobarak** [comment]
> > You have only unit tested the getDrivingFaultSumCountCat for the subcategories, there's a lot of other functions added which need unit testing and should be written
> 
> No, just no

Not valid anymore

**Review by appleabs — APPROVED** [review]

**Review by TomBillingtonUK — APPROVED** [review]

**Pull request merged**